docs/fpc-key-dist.puml

```diff
@@ -1,97 +1,245 @@
+    note left
+        enclave signs the key material it distributes together with the
+        sender and intended recipient identity
+    end note
+
+  return ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1 /'ECC1 -> Enclave1'/
+  return ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1 /'ECC_Manager1 -> ECC1'/
+  return ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1 /'Admin1 -> ECC_Manager1'/
+
+  note right Admin1
+    Key distribution is in progress. We need a means to transfer the message with
+    the encrypted keys to the recipient enclave. The next step uses the ledger
+    itself for this data transfer. This additionally allows to log the two phases
+    of the key distribution: (1) the sender enclave produced the encrypted data blob;
+    (2) the recipient enclave correctly received it. This can be useful when dealing
+    with enclaves of two different orgs.
+  end note
+
+  Admin1        -> ERCC1            ++: Start_KeyDist(ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1)
+  ERCC1         -> ERCC1            :   check ECC1_S_PK, ECC2_s_PK are registered and can endorse
+  ERCC1         -> ERCC1            :   check SIG_ECC1
+  ERCC1         -> ERCC1            :   put_state({"keydist_started", ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1})
+  return /'Admin1 -> ERCC1'/
+end group
+
+group complete key distribution
+  Admin2        -> ECC_Manager2     ++: ConfirmCCKeyDist(CC_name, CC_ver, CH_id)
+  ECC_Manager2  -> ECC2             ++: ConfirmCCKeyDist(CC_name, CC_ver, CH_id)
+  ECC2          -> Enclave2         ++: ConfirmCCKeyDist(CC_name, CC_ver, CH_id)
+  Enclave2      -> ERCC2            ++: GetKeyDistStart(ECC2_S_PK)
+  return keydist_start_message <- {"keydist_started", ECC1_S_PK, ECC2_S_PK, m, CC_PK, SIG_ECC1} /'Enclave2 -> ERCC2'/
+  Enclave2      -> TLCC_Enclave2    ++: check_committed_data(keydist_start_message)
+  return true /'Enclave2 -> TLCC_Enclave2'/
+    note left
+        if false, abort
+    end note
+  Enclave2      -> ERCC2            ++: GetRegistration(ECC1_S_PK)
+  return registration_message <- {CC_name, CC_ver, CH_id, org_id, ECC1_S_PK, ECC1_E_PK, attestation_report} /'Enclave2 -> ERCC2'/
+  Enclave2      -> TLCC_Enclave2    ++: check_committed_data(registration_message)
+  return true /'Enclave2 -> TLCC_Enclave2'/
+    note left
+        if false, abort
+    end note
+  Enclave2      -> Enclave2         :   verify attestation report
+  Enclave2      -> Enclave2         :   verify ECC1_S_PK matches in registration_message and keydist_start_message
+  Enclave2      -> Enclave2         :   verify SIG_ECC1
+  Enclave2      -> Enclave2         :   SEK, CC_SK <- decrypt(m) with ECC2_E_SK
+
+  Enclave2      -> Enclave2         :   SIG_ECC <- sign {CC_PK} with ECC2_S_SK
+  Enclave2      -> Enclave2         :   store SEK, CC_SK in sealed data
+  return ECC2_S_PK, SIG_ECC, CC_PK /'Enclave2 -> ECC2'/
+  return ECC2_S_PK, SIG_ECC, CC_PK /'ECC2 -> ECC_Manager2'/
```

Comment:
that's true, I have not repeated that because I think we should remove it (from this design chart).
The sealed data should be treated separately. First, we might not handle that now, just for the sake of simplification. Second, we could address it separately, for instance triggering a `save_sealed_data` each time there is an update.

Reply:
fixed